Tests/MapboxNavigationTests/ImageDownloaderTests.swift

```diff
@@ -135,4 +124,134 @@ class ImageDownloaderTests: XCTestCase {
         print("Succeeded after evaluating second condition \(spinCount) times.")
         XCTAssertTrue(callbackCalled)
     }
+
+    func testDownloadImageWithIncorrectUrl() {
+        var imageReturned: UIImage?
+        var dataReturned: Data?
+        var errorReturned: Error?
+        let imageDownloaded = expectation(description: "Image Downloaded")
+
+        let incorrectUrl = URL(fileURLWithPath: "/incorrect_url")
+        downloader.downloadImage(with: incorrectUrl) { (image, data, error) in
+            imageReturned = image
+            dataReturned = data
+            errorReturned = error
+            imageDownloaded.fulfill()
+        }
+        waitForExpectations(timeout: 10, handler: nil)
+
+        XCTAssertNil(imageReturned)
+        XCTAssertNil(dataReturned)
+        XCTAssertNotNil(errorReturned)
+    }
+
+    func testDownloadWith400StatusCode() {
+        var imageReturned: UIImage?
+        var dataReturned: Data?
+        var errorReturned: Error?
+        let imageDownloaded = expectation(description: "Image Downloaded")
+
+        let faultyUrl = URL(string: "https://www.mapbox.com/incorrect")!
```

Comment:
Not very reliable way of testing 4xx status code. Will extend the url spy with ability to emulate errors

Reply:
Done